pkg/util/kubernetes/apiserver/hpa_controller.go

```diff
@@ -386,8 +388,12 @@ func (h *AutoscalersController) deleteAutoscaler(obj interface{}) {
 
 	log.Debugf("Deleting Metrics from HPA %s/%s", deletedHPA.Namespace, deletedHPA.Name)
 	toDelete := hpa.Inspect(deletedHPA)
-	h.store.DeleteExternalMetricValues(toDelete)
-	h.queue.Done(deletedHPA)
+	log.Debugf("Deleting %s/%s from the local cache", deletedHPA.Namespace, deletedHPA.Name)
+	h.deleteFromLocalStore(toDelete)
+	if err := h.store.DeleteExternalMetricValues(toDelete); err != nil {
+		h.enqueue(deletedHPA)
+		return
+	}
```

Comment:
@clamoriniere are we not missing a h.queue.Forget here for the best case scenario ? To be safe.

Reply:
not need to `forget`, the `defer h.enqueue.Done()` in `h.processNext()` in enough. `Forget()` usually is used when you want to remove an item what is not currently process, or after `n` processing retry we remove it from the `workqueue.RateLimitingInterface`
So for instance in order to protect use to retry infinitely a item, we ca do in the `h.processNext()`
```
if h.queue.NumRequeues(key) > 10 {
    h.queue.Forget(key)
}
```